### ISSUE TITLE
Use containers to provide web applications.

### ISSUE THREAD

**Issue by JonTheNiceGuy** [lugorguk/test-deploy#7]
For example, wordpress or drupal? These can be entirely self-contained then, and only expose the nominated TCP ports needed for the web service.

Code to deploy docker is in the [lugorguk.docker](https://github.com/lugorguk/test-deploy/tree/main/roles/lugorguk.docker) role. There is no current code for any of the example sites.

**Comment by RalphCorderoy** [comment]
We should try to use containers, not necessarily Docker, to provide as many of the services as possible.

**Comment by gwestwood** [comment]
I agree with the use of containerisation of services, although I have no experience in this field to be able to offer any help or preference.

If this were to enable us to offer more options and easier deployment for LUGs (although, to prevent abuse by, say scammers, it should not be auto-created without an Admin involved).

**Comment by RalphCorderoy** [comment]
A friend at Red Hat that does container stuff day in, day out, likes https://podman.io/whatis.html as a Docker alternative. Could be worth a look by the time we move to containers.

**Comment by JonTheNiceGuy** [comment]
So, my "minimum viable product" plan for day 0 is to containerise Dokuwiki and static hosting "containers", which will *not* contain the content inside them. Instead, it'll use volumes, pointing to the underlying storage (hence #1). Over time we can then consider adding more applications (like, perhaps Wordpress, Drupal et. al) or allow lugmasters to propose their own containers.

I would not intend (on day 0) to have an automated deployment of anything... but I might use Ansible to deploy it :)

**Comment by JonTheNiceGuy** [comment]
> 
> 
> A friend at Red Hat that does container stuff day in, day out, likes https://podman.io/whatis.html as a Docker alternative. Could be worth a look by the time we move to containers.

Looks interesting. The Iron Sysadmin podcast, run now mostly by Red Hat people, recently did a good podcast on Podman. It does, indeed, sound interesting.
